.github/workflows/build-plugin-zip.yml

```diff
@@ -48,38 +141,62 @@ jobs:
           name: gutenberg-plugin
           path: ./gutenberg.zip
 
+      - name: Build release notes draft
+        if: ${{ needs.bump-version.outputs.new_version }}
+        env:
+          VERSION: ${{ needs.bump-version.outputs.new_version }}
+        run: |
+          IFS='.' read -r -a VERSION_ARRAY <<< "${VERSION}"
+          MILESTONE="Gutenberg ${VERSION_ARRAY[0]}.${VERSION_ARRAY[1]}"
+          npm run changelog -- --milestone="$MILESTONE" --unreleased > release-notes.txt
+          sed -ie '1,6d' release-notes.txt
+          if [[ ${{ needs.bump-version.outputs.new_version }} != *"rc"* ]]; then
+            # Include previous RCs' release notes, if any
+            CHANGELOG_REGEX="=\s[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?\s="
+            RC_REGEX="=\s${VERSION}(-rc\.[0-9]+)?\s="
+            awk "/${RC_REGEX}/ {found=1;print;next} /${CHANGELOG_REGEX}/ {found=0} found" changelog.txt >> release-notes.txt
+          fi
+
+      - name: Upload release notes artifact
+        if: ${{ needs.bump-version.outputs.new_version }}
+        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571 # v2.2.2
+        with:
+          name: release-notes
+          path: ./release-notes.txt
+
   create-release:
     name: Create Release Draft and Attach Asset
-    needs: build
+    needs: [bump-version, build]
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags/v')
     steps:
       - name: Set Release Version
         id: get_release_version
-        run: echo ::set-output name=version::$(echo $GITHUB_REF | cut -d / -f 3 | sed s/^v// | sed 's/-rc./ RC/' )
+        env:
+          VERSION: ${{ needs.bump-version.outputs.new_version }}
+        run: echo ::set-output name=version::$(echo $VERSION | cut -d / -f 3 | sed 's/-rc./ RC/' )
 
       - name: Download Plugin Zip Artifact
         uses: actions/download-artifact@4a7a711286f30c025902c28b541c10e147a9b843 # v2.0.8
         with:
           name: gutenberg-plugin
 
-      - name: Extract Changelog for Release
-        run: |
-          unzip gutenberg.zip changelog.txt
-          CHANGELOG_REGEX="/=\s[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?\s=/"
-          awk -i inplace "$CHANGELOG_REGEX"'{p++;next} p==2{exit} p>=1' changelog.txt
+      - name: Download Release Notes Artifact
+        uses: actions/download-artifact@4a7a711286f30c025902c28b541c10e147a9b843 # v2.0.8
+        with:
+          name: release-notes
 
       - name: Create Release Draft
         id: create_release
         uses: actions/create-release@0cb9c9b65d5d1901c1f53e5e66eaf4afd303e70e # v1.1.4
```

Comment:
Any thoughts on this being unmaintained now?

https://github.com/actions/create-release

Reply:
Yeah, I saw that the other day. I'm thinking to eventually replace it by an [`actions/script`](https://github.com/actions/github-script)-based step.

Reply:
Filed an issue to keep track: https://github.com/WordPress/gutenberg/issues/29946